test/rqg/mzcompose.py

```diff
@@ -193,40 +227,35 @@ def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
 
 
 def run_workload(c: Composition, args: argparse.Namespace, workload: Workload) -> None:
+    def materialize_image(tag: str | None) -> str | None:
+        return f"materialize/materialized:{tag}" if tag else None
+
+    # A list of psql-compatible services participating in the test
     participants: list[Service] = [
         Materialized(
             name="mz_this",
             ports=["16875:6875", "16876:6876", "16877:6877", "16878:6878"],
-            image=f"materialize/materialized:{args.this_tag}"
-            if args.this_tag
-            else None,
+            image=materialize_image(args.this_tag),
             use_default_volumes=False,
         ),
     ]
 
+    # A list of psql URLs for dataset initialization
     psql_urls = ["postgresql://materialize@mz_this:6875/materialize"]
 
-    if args.other_tag == "common-ancestor":
```

Comment:
Moved this to a `StoreOtherTag(argparse.Action)` implementation (used with `action=StoreOtherTag` when configuring the `argparse` argument).